src/box/applier.cc

```diff
@@ -2842,7 +2844,8 @@ applier_on_state_f(struct trigger *trigger, void *event)
 	/* Wake up waiter */
 	fiber_cond_signal(&on_state->wakeup);
 
-	applier_pause(applier);
+	if (applier->state == on_state->desired_state)
+		applier_pause(applier);
```

Comment:
Why not always call the trigger from the applier fiber then, even in case the fiber is stopped with `applier_stop`?